src/components/Card/Card.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect } from 'react';
+import { CardProps, Character } from '../../interfaces/interfaces';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  selectCharacter,
+  unselectCharacter,
+} from '../../store/selectedItemsSlice';
+import { RootState } from '../../store/store';
+import { useSearchCharactersQuery } from '../../api/starWarsApi';
+import { setLoading } from '../../store/uiSlice';
+import styles from './Card.module.css';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import Loading from '../Loader/Loader';
+import useTheme from '../../hooks/useTheme';
+
+const Card: React.FC<CardProps> = ({ character }) => {
+  const navigate = useNavigate();
+  const { theme } = useTheme();
+  const [searchParams] = useSearchParams();
+  const dispatch = useDispatch();
+
+  const selectedCharacters = useSelector(
+    (state: RootState) => state.selectedItems.selectedCharacters
+  );
+
+  const isLoading = useSelector((state: RootState) => state.ui.isLoading);
+
+  const searchTerm = searchParams.get('query') || '';
+  const page = searchParams.get('page') || '1';
+
+  const { error, isFetching } = useSearchCharactersQuery({
+    searchTerm,
+    page: parseInt(page),
+  });
+
+  useEffect(() => {
+    const selected = JSON.parse(localStorage.getItem('selectedItems') || '[]');
+    selected.forEach((char: Character) => {
```

Comment:
If selected contains a large number of items, forEach calls dispatch multiple times synchronously, which may cause performance issues - so it will be better to create a separate action, f.e. `selectMultipleCharacters` that handles multiple characters at once